athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/GcsSchemaUtils.java

```diff
@@ -0,0 +1,117 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connectors.gcs;
+
+import com.amazonaws.athena.connector.lambda.data.SchemaBuilder;
+import com.amazonaws.athena.connectors.gcs.common.StorageLocation;
+import com.amazonaws.athena.connectors.gcs.storage.StorageMetadata;
+import com.amazonaws.athena.connectors.gcs.storage.datasource.StorageTable;
+import com.amazonaws.services.glue.model.Table;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.CLASSIFICATION_GLUE_TABLE_PARAM;
+import static com.amazonaws.athena.connectors.gcs.GcsUtil.isFieldTypeNull;
+
+public class GcsSchemaUtils
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(GcsSchemaUtils.class);
+
+    protected GcsSchemaUtils()
+    {
+    }
+
+    /**
+     * Builds the table schema based on the provided field by the retrieved instance of {@link StorageTable}
+     *
+     * @param datasource An instance of {@link StorageMetadata}
+     * @param table      Glue table object
+     * @return An instance of {@link Schema}
+     */
+    public static Schema buildTableSchema(StorageMetadata datasource, Table table) throws Exception
+    {
+        SchemaBuilder schemaBuilder = SchemaBuilder.newBuilder();
+        String locationUri = table.getStorageDescriptor().getLocation();
+        StorageLocation storageLocation = StorageLocation.fromUri(locationUri);
+        Optional<StorageTable> optionalStorageTable = datasource.getStorageTable(storageLocation.getBucketName(), storageLocation.getLocation(), table.getParameters().get(CLASSIFICATION_GLUE_TABLE_PARAM));
+        if (optionalStorageTable.isPresent()) {
+            StorageTable sTable = optionalStorageTable.get();
+            LOGGER.debug("Schema Fields\n{}", sTable.getFields());
+            for (Field field : sTable.getFields()) {
+                if (isFieldTypeNull(field)) {
+                    field = Field.nullable(field.getName().toLowerCase(), Types.MinorType.VARCHAR.getType());
+                }
+                else {
+                    field = new Field(field.getName().toLowerCase(), new FieldType(true, field.getType(), field.getDictionary(), field.getMetadata()), field.getChildren());
```

Comment:
Should we pass `field.isNullable()` instead of always true here? What was the reason we set alwasys true

Reply:
incorporated suggested changes.